### PR TITLE
Include allowed roles in security scheme scopes when OpenApi version is 3.1.0 or greater

### LINKED ISSUE #37199: Support @RolesAllowed values populated to openapi schema now that 3.1.0 spec supports it
### Description

**Issue**:

`@RolesAllowed` annotation values are not populated to openapi schema.yaml, As a workaround, If I add 
`@SecurityRequirement(name = "BearerJWTScheme", scopes = { "roleA", "roleB" }) `I see it populated properly but it doesnt get picked automatically even with [quarkus.smallrye-openapi.auto-add-security-requirement](https://quarkus.io/guides/openapi-swaggerui#quarkus-smallrye-openapi_quarkus.smallrye-openapi.auto-add-security-requirement) property configured to true.

**Observation:**

On further analysis, I see it was allowed only for "oauth2" or "openIdConnect" security scheme due to the fact, it was not supported for other security schemes till open api 3.0.x version, as outlined in this issue, https://github.com/quarkusio/quarkus/issues/27373

**Expectation:**

Support Open API spec [3.1.0](https://github.com/OAI/OpenAPI-Specification/blob/3.1.0/versions/3.1.0.md#security-requirement-object)

As explained in spec,

Each name MUST correspond to a security scheme which is declared in the [Security Schemes](https://github.com/OAI/OpenAPI-Specification/blob/3.1.0/versions/3.1.0.md#componentsSecuritySchemes) under the [Components Object](https://github.com/OAI/OpenAPI-Specification/blob/3.1.0/versions/3.1.0.md#componentsObject). If the security scheme is of type "oauth2" or "openIdConnect", then the value is a list of scope names required for the execution, and the list MAY be empty if authorization does not require a specified scope. For other security scheme types, the array MAY contain a list of role names which are required for the execution, but are not otherwise defined or exchanged in-band.

Sample on how it would like in [openapi spec](https://github.com/OAI/OpenAPI-Specification/pull/2515/files)

```
openapi: 3.1.0
info:
  title: Non-oAuth Scopes example
  version: 1.0.0
paths:
  /users:
    get:
      security:
        - bearerAuth:
            - 'read:users'
            - 'public'
components:
  securitySchemes:
    bearerAuth:
      type: http
      scheme: bearer
      bearerFormat: jwt
      description: 'note: non-oauth scopes are not defined at the securityScheme level'
```

### Implementation ideas

**Solution:**

I would like to raise a PR to fix this by removing the condition here, https://github.com/quarkusio/quarkus/blob/main/extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/filter/AutoRolesAllowedFilter.java#L114

Since the open api version is configurable via [quarkus.smallrye-openapi.open-api-version](https://quarkus.io/guides/openapi-swaggerui#quarkus-smallrye-openapi_quarkus.smallrye-openapi.open-api-version).

But it might affect the users who are not migrated to new open api version and they might get undesired schema output.

To fix this properly, I see two options,
1. We could check against the configured open api version and disable the condition but not sure whether quarkus team wants to maintain code to support multiple versions or 
2. Wait for [smallrye-openapi to start supporting 3.1.0 by default](https://github.com/smallrye/smallrye-open-api/blob/main/core/src/main/java/io/smallrye/openapi/api/constants/OpenApiConstants.java#L119) via issue mentioned on [microprofile-openapi](https://github.com/eclipse/microprofile-open-api/issues/333).

Looking for direction from Quarkus team on how to proceed on this ?

### PR DESCRIPTION
**Pull request by michalvavrik**
- closes: https://github.com/quarkusio/quarkus/issues/37199

### CODE REVIEW

**Comment by michalvavrik** [comment]
Hey @MikeEdgar I can't request review from you for some reason, sorry, please have a look.

**Review by phillip-kruger — APPROVED** [review]
LGTM

**Review by MikeEdgar — COMMENTED** [review]
Looks good, thanks @michalvavrik

File: extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java [hunk 1]
```diff
@@ -1169,4 +1169,9 @@ private List<String> getResourceFiles(Path resourcePath, Path target) {
         }
         return filenames;
     }
+
+    private static boolean isOpenApi_3_1_0_OrGreater(SmallRyeOpenApiConfig config) {
+        final String openApiVersion = config.openApiVersion.orElse(null);
+        return openApiVersion == null || (!openApiVersion.startsWith("2") && !openApiVersion.startsWith("3.0"));
```
Comment (gsmet) [inline extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java:1175]: I'm having a look at this one as I was planning to backport it. When the version is not defined, we consider it's the most recent version?

Comment (michalvavrik) [reply extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java:1175]: When the version is not defined, it's the latest supported according to my tests. I am not sure if you can backport that as based on this comment https://github.com/quarkusio/quarkus/issues/37199#issuecomment-2302005662 I think it requires https://github.com/quarkusio/quarkus/pull/44055. Better wait for answers from Mike or Phillip.

Comment (MikeEdgar) [reply extensions/smallrye-openapi/deployment/src/main/java/io/quarkus/smallrye/openapi/deployment/SmallRyeOpenApiProcessor.java:1175]: smallrye-open-api will default to the latest depending on the SR version - so `3.0.3` prior to release 4.0, and now `3.1.0` as of 4.0 (#44055). I suspect that may be an issue for backporting, as @michalvavrik already said.

**Pull request merged**